.circleci/workflows.yml

```diff
@@ -0,0 +1,1151 @@
+          steps:
+            - checkout
+            - *restore_venv_cache
+            - *build
+            - &attach_generated_sql
+              attach_workspace:
+                at: /tmp/workspace
+            - &copy_staged_sql
+              run:
+                name: Move sql deployed on stage into place
+                command: |
+                  rm -rf sql/
+                  cp -r /tmp/workspace/staged-generated-sql/sql sql
+                  rm -rf tests/
+                  cp -r /tmp/workspace/staged-generated-sql/tests tests
+            - run:
+                name: Run SQL tests
+                command: |
+                  if [ -n "$CIRCLE_PR_NUMBER" ]; then
+                    echo "Cannot pass creds to forked PRs," \
+                      "so skipping routine and SQL tests"
+                  else
+                    PATH="venv/bin:$PATH" script/entrypoint -m sql -n 8
+                  fi
+      - run: exit 0
+  dry-run-sql:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
+          steps:
+            - checkout
+            - *restore_venv_cache
+            - *build
+            - *attach_generated_sql
+            - *copy_staged_sql
+            - run:
+                name: Dry run queries
+                # yamllint disable rule:line-length
+                # Dry runs on PRs are executed on sql/bigquery-etl-integration-test
+                # Artifacts (queries, views, UDFs) that are changed will be moved to the
+                # bigquery-etl-integration-test folder and deployed to the corresponding
+                # project. This ensures that dry runs can be executed before changes
+                # have been deployed to prod. (bigquery-etl-integration-test is treated
+                # as a stage environment)
+                command: |
+                  if [ "$CIRCLE_BRANCH" = main ]; then
+                    echo "Check dry run for all queries because branch is" \
+                      "$CIRCLE_BRANCH"
+                    PATHS=sql
+                  elif git log --format=%B --no-merges -n 1 |
+                      grep -qF '[run-tests]'; then
+                    echo "Check dry run for all queries because [run-tests] in" \
+                      "commit message"
+                    PATHS=sql
+                  else
+                    PATHS="sql/bigquery-etl-integration-test"
+                  fi
+                  echo $PATHS
+                  PATH="venv/bin:$PATH" script/bqetl dryrun --validate-schemas $PATHS
+                # yamllint enable rule:line-length
+      - run: exit 0
+  validate-metadata:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
```

Comment:
I wouldn't think that routines changing would make any difference for validating query metadata, except the `bqetl query validate` command being run is currently also dry-running the queries (even though that's kind of duplicative of the `dry-run-sql` job) which routines could affect.